### PR TITLE
Use odo to discover local components

### LINKED ISSUE #178: Support for component configuration through 'odo config' calls
[kind/Feature]

## Which functionality do you think we should add?

See https://github.com/redhat-developer/vscode-openshift-tools/issues/1461
or https://issues.redhat.com/browse/JBIDE-27124

### PR DESCRIPTION
**Pull request by jeffmaury**
Fixes #178

Signed-off-by: Jeff MAURY <jmaury@redhat.com>

## Was the change discussed in an issue?
fixes #???

### CODE REVIEW

File: src/main/java/org/jboss/tools/intellij/openshift/tree/application/ApplicationTreeModel.java [hunk 1]
```diff
@@ -120,21 +88,18 @@ public void moduleRemoved(@NotNull Project project, @NotNull Module module) {
...
 
     private void addContext(VirtualFile modulePathFile) {
         if (modulePathFile != null && modulePathFile.isValid()) {
             try {
-                VirtualFile file = modulePathFile.findFileByRelativePath(ODO_CONFIG_YAML);
-                if (file != null && file.isValid()) {
-                    LocalConfig config = LocalConfig.load(new File(file.getPath()).toPath().toUri().toURL());
-                    addContextToSettings(modulePathFile.getPath(), config.getComponentSettings());
-                }
+                List<ComponentDescriptor> descriptorss = ROOT.getOdo().discover(modulePathFile.getPath());
```
Comment (sbouchet) [inline src/main/java/org/jboss/tools/intellij/openshift/tree/application/ApplicationTreeModel.java:101]: is this double S intended ? if no, please correct the typo.

Comment (jeffmaury) [reply src/main/java/org/jboss/tools/intellij/openshift/tree/application/ApplicationTreeModel.java]: Good catch

File: src/test/java/org/jboss/tools/intellij/openshift/actions/component/CreateURLActionTest.java [hunk 2]
```diff
@@ -23,4 +23,14 @@ public AnAction getAction() {
     protected void verifyPushedComponent(boolean visible) {
         assertTrue(visible);
     }
+
+    @Override
+    protected void verifyNotPushedComponent(boolean visible) {
+        assertTrue(visible);;
```
Comment (sbouchet) [inline src/test/java/org/jboss/tools/intellij/openshift/actions/component/CreateURLActionTest.java:29]: please remove extra semicolon.

File: src/test/java/org/jboss/tools/intellij/openshift/actions/component/CreateURLActionTest.java [hunk 3]
```diff
@@ -23,4 +23,14 @@ public AnAction getAction() {
...
+    }
+
+    @Override
+    protected void verifyNoContextComponent(boolean visible) {
+        assertTrue(visible);;
```
Comment (sbouchet) [inline src/test/java/org/jboss/tools/intellij/openshift/actions/component/CreateURLActionTest.java:34]: same as above, please remove extra semicolon.

**Review by sbouchet — APPROVED** [review]
LGTM. good work ! 
tested with remote openshift cluster with success.

**Pull request merged**